src/lib/proxy/proxyConfig.ts

```
@@ -0,0 +1,513 @@
+/**
+ * Walk the resolved config tree and warn about any remaining `${VAR}`
+ * placeholders that were not resolved.  Returns the list of unresolved
+ * variable names so callers can decide whether to abort.
+ */
+function warnUnresolvedPlaceholders(obj: unknown, path = ""): string[] {
+  const unresolved: string[] = [];
+  if (typeof obj === "string") {
+    // Reset global regex state before matching
+    ENV_VAR_PATTERN.lastIndex = 0;
+    let match: RegExpExecArray | null;
+    while ((match = ENV_VAR_PATTERN.exec(obj)) !== null) {
```

Biome (2.4.7)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)